troubleshooting/known_issues/SBPS_boot_fail.md

```diff
@@ -0,0 +1,150 @@
+```
+
+An example run of the command listing the problematic `manifest.json` is as follows:
+
+```bash
+# for f in /var/lib/cps-local/boot-images/*/manifest.json; do \
+     echo $f etag=$(cat $f | jq '.artifacts[] | select(.type == "application/vnd.cray.image.rootfs.squashfs").link.etag' );  \
+ done | grep etag=null
+/var/lib/cps-local/boot-images/3f89b76d-7578-4ac3-8891-0f8b4335b06d/manifest.json etag=null
+```
+
+Looking at the `manifest.json`:
+
+```bash
+ncn-w006:~ # cat /var/lib/cps-local/boot-images/3f89b76d-7578-4ac3-8891-0f8b4335b06d/manifest.json
+{
+   "artifacts" : [
+      {
+         "link" : {
+            "path" : "s3://boot-images/3f89b76d-7578-4ac3-8891-0f8b4335b06d/compute.squashfs",
+            "type" : "s3"
+         },
+         "md5" : "fa69a6665124629bfb1d74bfa3e4b7de",
+         "type" : "application/vnd.cray.image.rootfs.squashfs"
+      }
+   ],
+   "created" : "20240815093713",
+   "version" : "1.0"
+}
+```
+
+The "link" section has no `etag` field.
+
+Step 2: Update the manifest with `etag` and update the same in `s3`
+
+Example `manifest.json` update with `etag` obtained from `s3` and update the same in `s3`:
+
+```bash
+ncn-m001:~ # MANIFEST=$( cray ims images describe "3f89b76d-7578-4ac3-8891-0f8b4335b06d"  --format json | jq -r .link.path )
+ncn-m001:~ # echo $MANIFEST
+s3://boot-images/3f89b76d-7578-4ac3-8891-0f8b4335b06d/manifest.json
+
+ncn-m001:~ # cray artifacts get $( echo $MANIFEST | sed -e 's/^s3:\/\///g' -e 's/boot-images\//boot-images /g' ) manifest.json
+ncn-m001:~ # cat manifest.json
+{ "artifacts" : [ { "link" : { "path" : "s3://boot-images/3f89b76d-7578-4ac3-8891-0f8b4335b06d/compute.squashfs", "type" : "s3" }, "md5" : "fa69a6665124629bfb1d74bfa3e4b7de", "type" : "application/vnd.cray.image.rootfs.squashfs" } ], "created" : "20240815093713", "version" : "1.0" }
+ncn-m001:~ # cat manifest.json | jq -r '.artifacts[] | select(.type == "application/vnd.cray.image.rootfs.squashfs").link.path' | sed -e 's/^s3:\/\///g' -e 's/boot-images\//boot-images /g'
+boot-images 3f89b76d-7578-4ac3-8891-0f8b4335b06d/compute.squashfs
+
+ncn-m001:~ # cray artifacts describe $( cat manifest.json | jq -r '.artifacts[] | select(.type == "application/vnd.cray.image.rootfs.squashfs").link.path' | sed -e 's/^s3:\/\///g' -e 's/boot-images\//boot-images /g' ) --format json
+{
+  "artifact": {
+    "AcceptRanges": "bytes",
+    "LastModified": "2024-08-15T14:45:58+00:00",
+    "ContentLength": 22095331328,
+    "ETag": "\"f4153fe0a18cfb2205fb002d81a23ab2-2634\"",
+    "ContentType": "binary/octet-stream",
+    "Metadata": {
+      "md5sum": "fa69a6665124629bfb1d74bfa3e4b7de"
+    }
+  }
+}
+ncn-m001:~ # cray artifacts describe $( cat manifest.json | jq -r '.artifacts[] | select(.type == "application/vnd.cray.image.rootfs.squashfs").link.path' | sed -e 's/^s3:\/\///g' -e 's/boot-images\//boot-images /g' ) --format json | jq -r '.artifact.ETag' | tr -d '"'
+f4153fe0a18cfb2205fb002d81a23ab2-2634
+ncn-m001:~ # cat manifest.json

```

Comment:
You never give an explicit instruction to update the manifest with the etag. You could say this.

Now update the manifest.json file with the newly discovered `etag` as show below.